src/main/java/com/synopsys/integration/alert/channel/jira/server/JiraServerProperties.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.jira.server;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.alert.channel.jira.server.descriptor.JiraServerDescriptor;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.persistence.accessor.FieldAccessor;
+import com.synopsys.integration.jira.common.rest.JiraHttpClient;
+import com.synopsys.integration.jira.common.server.configuration.JiraServerRestConfig;
+import com.synopsys.integration.jira.common.server.configuration.JiraServerRestConfigBuilder;
+import com.synopsys.integration.jira.common.server.service.JiraServerServiceFactory;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+public class JiraServerProperties {
+    private final String url;
+    private final String password;
+    private final String username;
+
+    public JiraServerProperties(FieldAccessor fieldAccessor) {
+        url = fieldAccessor.getStringOrNull(JiraServerDescriptor.KEY_SERVER_URL);
+        password = fieldAccessor.getStringOrNull(JiraServerDescriptor.KEY_SERVER_PASSWORD);
+        username = fieldAccessor.getStringOrNull(JiraServerDescriptor.KEY_SERVER_USERNAME);
+    }
+
+    public JiraServerProperties(String url, String password, String username) {
+        this.url = url;
+        this.password = password;
+        this.username = username;
+    }
+
+    public JiraServerRestConfig createJiraServerConfig() throws AlertException {
+        JiraServerRestConfigBuilder jiraServerConfigBuilder = new JiraServerRestConfigBuilder();
+        String jiraUrl = url;
+        if (!StringUtils.endsWith(url, "/jira")) {
```

Comment:
We force the URL to end in '/jira'? For what reason?

Reply:
in order to install the indexing app in a jira server the URL must have /jira in the path.  I can make a AppServices class in int-jira-common that extends PluginManagerService that will create the url with the "/jira" in the path.  The /jira in the path is used as the context to be able to install apps into jira server.  Otherwise you get 404 errors 
Jira cloud does not require "/jira" in the path.  I will insulate Alert from that knowledge.

Reply:
I removed this code.  We'll just expect the URL to have /jira in it when they configure the channel.